interfaces/builtin/serial_port.go

```diff
@@ -111,22 +170,54 @@ func (iface *SerialPortInterface) PermanentPlugSnippet(plug *interfaces.Plug, se
 func (iface *SerialPortInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
 	switch securitySystem {
 	case interfaces.SecurityAppArmor:
-		return []byte(fmt.Sprintf("%s rwk,\n", iface.path(slot))), nil
-	case interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev, interfaces.SecurityMount:
+		if iface.hasUsbAttrs(slot) {
+			// This apparmor rule must match serialDeviceNodePattern
+			// UDev tagging and device cgroups will restrict down to the specific device
+			return []byte("/dev/tty[A-Z]{,[A-Z],[A-Z][A-Z]}[0-9]{,[0-9],[0-9][0-9]} rw,\n"), nil
+		}
```

Comment:
I think you left out:
```
else {
            return []byte(fmt.Sprintf("%s rw,\n", iface.path(slot))), nil
}
```

We want the glob when we have the usb attributes but we want the specified static path when we do not.